src/radix.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+"""File implements a radix sort method."""
+from __future__ import unicode_literals
+from collections import OrderedDict
+from collections import deque
+import random
+import timeit
+
+
+def radix_sort(sort_list):
+    """Sorts input based on the least significant digits of the data first
+    and cascades upward until the input has been sorted by it's most
+    signifiant digit.
+
+    Worked with Steven and Victor writing this function."""
+    if not isinstance(sort_list, list):
+        raise TypeError("Radix sort only accepts inputs inside a Python list.")
+    for item in sort_list:
+        if not isinstance(item, int):
+            raise TypeError("All the items in your list must be integers.")
+    biggest = max(sort_list)
+    num_sorts = len(str(biggest))
+    sort_list = [str(item).zfill(num_sorts) for item in sort_list]
```

Comment:
This will work, but is there another way you can make your sort work without the extra overhead of iterating across the whole list twice, converting ints to strs, then back to ints at the end?

There is a mathematical way you can determine how many places an integer has -- see if you can figure it out.